### PR TITLE
COMMONS-29 Sort elements in SelectEObjectDialog

### PR DESCRIPTION
**Pull request by seiferma**
This PR addresses [COMMONS-29](https://jira.palladio-simulator.com/browse/COMMONS-29). The implementation is straight forward but I might have missed cases where ordering is not wanted. However, this should be discussed in the issue.

### CODE REVIEW

File: bundles/de.uka.ipd.sdq.dialogs/src/de/uka/ipd/sdq/dialogs/selection/SelectEObjectDialog.java [hunk 1]
```diff
@@ -85,11 +89,36 @@ public SelectEObjectDialog(final Shell parentShell, final String resourceName, f
...
+    private static ViewerComparator createViewerComparator() {
+        return new ViewerComparator() {
+
+            private boolean isRootObject(Object o) {
+                Optional<EObject> eobject = Optional.ofNullable(o).filter(EObject.class::isInstance)
+                        .map(EObject.class::cast);
+                if (eobject.isPresent()) {
+                    return eobject.get().eContainer() == null;
+                }
+                return false;
```
Comment (sdkrach) [inline bundles/de.uka.ipd.sdq.dialogs/src/de/uka/ipd/sdq/dialogs/selection/SelectEObjectDialog.java:107]: you could boil this down to a .filter() and subsequent .isPresent()

Comment (seiferma) [reply bundles/de.uka.ipd.sdq.dialogs/src/de/uka/ipd/sdq/dialogs/selection/SelectEObjectDialog.java:107]: I do not think that this is possible using a filter step without the assumption that `o` is an EObject:
* `eobject.filter(e -> e.eContainer() == null).isPresent()` will yield true even if `o` is no EObject, which means it cannot be an root element. Therefore, it is not correct.
* `!eobject.filter(e -> e.eContainer != null).isPresent()` also yields a wrong result if `o` is no EObject.

However, I think a map statement would yield the correct result. I did not come to this solution while writing the code but I will integrate it now:
`eobject.map(e -> e.eContainer == null).orElse(false)`

Comment (sdkrach) [reply bundles/de.uka.ipd.sdq.dialogs/src/de/uka/ipd/sdq/dialogs/selection/SelectEObjectDialog.java]: `eobject.filter(e -> e.eContainer() == null).isPresent()` will yield true, iff `eobject.isPresent()` yields true (therefore o is an EObject) and `((EObject)o).eContainer==null`. Or am I mistaken?

Comment (seiferma) [reply bundles/de.uka.ipd.sdq.dialogs/src/de/uka/ipd/sdq/dialogs/selection/SelectEObjectDialog.java]: My statements made sense when I thought about it but you are right ;)

**Pull request merged**